### PR TITLE
od2od The format <-> binary file converter. 

### PR DESCRIPTION
**Pull request by ajm143**
Tried to test this for robustness and document its limitations. Now works for OME, DOME, PDOS and ELNES.

To do -- make some tests in the test suite.

### CODE REVIEW

File: optados/src/electronic.f90 [hunk 1]
```diff
@@ -1167,9 +1244,12 @@ subroutine elec_pdos_read
         enddo
         call comms_send(pdos_weights(1, 1, 1, 1), pdos_mwab%norbitals*pdos_mwab%nbands* &
                         nspins*num_kpoints_on_node(inodes), inodes)
-      end do
+     end do
+
+     write(*,*) num_kpoints_on_node(0)
```
Comment (jryates) [inline optados/src/electronic.f90]: is this a debug line?

File: optados/src/electronic.f90 [hunk 2]
```diff
@@ -1167,9 +1244,12 @@ subroutine elec_pdos_read
...
       do ik = 1, num_kpoints_on_node(0)
-        ! The kpoint number, followed by the kpoint-vector
+         ! The kpoint number, followed by the kpoint-vector
+              write(*,*) "loop", ik
```
Comment (jryates) [inline optados/src/electronic.f90]: is this a debug line?

**Comment by jryates** [comment]
This is now failing due to an issue with build.f90
 This is because the script that writes build.f90 uses "git describe --tags" - this fails as the repository has no tags (perhaps @ajm143 has some uncommitted tags in his repository - hence it worked locally).
 If I swap the line for "git describe --always" it builds ok. But perhaps the answer is to commit some tags.

**Comment by ajm143** [comment]
Oh -- I seem to have got two pull requests merged together.  Let me unpick and try again. There will be three separate things happening, and they deserve a pull request each:

(1) od2od file converter
(2) The --version command.
(3) First attempt at automatic documentation.

Sorry,
A.

**Comment by jryates** [comment]
Andrew - I think you still need to strip the debug print statements noted above.

**Comment by ajm143** [comment]
Hi Jonathan -- 

Yes. I don't think I've got this right yet.  At the moment I've got optados/optados ajm143/optados, ~/src/optados and ~/tmp/optados all in slightly different states.  For what I believe are the right reasons. 

I'm not yet competent to achieve what I need to with an optically minimal number of moves!  But I think I'm learning.

**Comment by jryates** [comment]
It gets a bit messy working with a live pull request! Push all you need to, then drop me a note when you want me to check it over and merge.

**Comment by ajm143** [comment]
Ta. I'm sure there are more elegant ways to do what I'm doing -- but at the moment I've only got one wheel on my wagon...

**Comment by ajm143** [comment]
Ok -- the two debug line comments are now "outdated" so I think I've got this pull request right. Can you check and merge if you're happy.

**Review by jryates — APPROVED** [review]

**Pull request merged**